ir/shared/src/main/scala/org/scalajs/ir/Serializers.scala

```diff
@@ -1493,6 +1531,238 @@ object Serializers {
+          val newName = MethodIdent(NoArgConstructorName)(method.name.pos)
+          val newBody = ApplyStatically(ApplyFlags.empty.withConstructor(true),
+              thisJLClass, ObjectClass, newName, Nil)(NoType)
+          MethodDef(method.flags, newName, method.originalName,
+              Nil, NoType, Some(newBody))(
+              method.optimizerHints, method.version)
+        } else if (methodSimpleNameString == "newArrayOfThisClass") {
+          val newName = MethodIdent(MethodName(methodName.simpleName,
+              List(IntRef), ClassRef(ObjectClass)))(method.name.pos)
+          val lengthParam = ParamDef(LocalIdent(LocalName("length")),
+              NoOriginalName, IntType, mutable = false)
+          val newBody = BinaryOp(BinaryOp.Class_newArray, thisJLClass, lengthParam.ref)
+          MethodDef(method.flags, newName, method.originalName,
+              List(lengthParam), AnyType, Some(newBody))(
+              method.optimizerHints.withInline(true), method.version)
+        } else {
+          def argRef = method.args.head.ref
+
+          val newBody: Tree = methodSimpleNameString match {
+            case "getName"          => UnaryOp(UnaryOp.Class_name, thisJLClass)
+            case "isPrimitive"      => UnaryOp(UnaryOp.Class_isPrimitive, thisJLClass)
+            case "isInterface"      => UnaryOp(UnaryOp.Class_isInterface, thisJLClass)
+            case "isArray"          => UnaryOp(UnaryOp.Class_isArray, thisJLClass)
+            case "getComponentType" => UnaryOp(UnaryOp.Class_componentType, thisJLClass)
+            case "getSuperclass"    => UnaryOp(UnaryOp.Class_superClass, thisJLClass)
+
+            case "isInstance"       => BinaryOp(BinaryOp.Class_isInstance, thisJLClass, argRef)
+            case "isAssignableFrom" => BinaryOp(BinaryOp.Class_isAssignableFrom, thisJLClass, argRef)
+            case "cast"             => BinaryOp(BinaryOp.Class_cast, thisJLClass, argRef)
+
+            case _ =>
+              /* Unfortunately, some of the other methods directly referred to
+               * `this.data["name"]`, instead of building on `this.getName()`.
+               * We must replace those occurrences with a `Class_name` as well.
+               */
+              val transformer = new Transformers.Transformer {
+                override def transform(tree: Tree, isStat: Boolean): Tree = tree match {
+                  case JSSelect(_, StringLiteral("name")) =>
+                    implicit val pos = tree.pos
+                    UnaryOp(UnaryOp.Class_name, thisJLClass)
+                  case _ =>
+                    super.transform(tree, isStat)
+                }
+              }
+              transformer.transform(method.body.get, isStat = method.resultType == NoType)
+          }
+
+          MethodDef(method.flags, method.name, method.originalName,
+              method.args, method.resultType, Some(newBody))(
+              method.optimizerHints.withInline(true), method.version)
```

Comment:
It seems marking `getSimpleName` and `computeCachedSimpleNameBestEffort` inline flagged too (am I correct?), is that okay? I don't see those methods`@inline` in Class.scala.

Reply:
Good catch! Fixed.